src/Microsoft.ComponentDetection.Detectors/pub/PubComponentDetector.cs

```diff
@@ -0,0 +1,67 @@
+namespace Microsoft.ComponentDetection.Detectors.Pub;
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.ComponentDetection.Contracts;
+using Microsoft.ComponentDetection.Contracts.Internal;
+using Microsoft.ComponentDetection.Contracts.TypedComponent;
+using Microsoft.Extensions.Logging;
+using YamlDotNet.Serialization;
+
+public class PubComponentDetector : FileComponentDetector
+{
+    public PubComponentDetector(
+        IComponentStreamEnumerableFactory componentStreamEnumerableFactory,
+        IObservableDirectoryWalkerFactory walkerFactory,
+        ILogger<PubComponentDetector> logger)
+    {
+        this.ComponentStreamEnumerableFactory = componentStreamEnumerableFactory;
+        this.Scanner = walkerFactory;
+        this.Logger = logger;
+    }
+
+    public override string Id => "pub";
+
+    public override IEnumerable<string> Categories => new[] { Enum.GetName(typeof(DetectorClass), DetectorClass.Pub) };
+
+    public override IEnumerable<ComponentType> SupportedComponentTypes { get; } = new[] { ComponentType.Pub };
+
+    public override int Version => 1;
+
+    public override IList<string> SearchPatterns => new List<string> { "pubspec.lock" };
+
+    protected override async Task OnFileFoundAsync(ProcessRequest processRequest, IDictionary<string, string> detectorArgs)
+    {
+        using var reader = new StreamReader(processRequest.ComponentStream.Stream);
+        var text = await reader.ReadToEndAsync();
```

Comment:
If we fail to read the file CD will crash, lets move this into the `try` block below.